### ISSUE TITLE
TNSNativeSource gets duplicated in Xcode projects

### ISSUE THREAD

**Issue by jvaronenka** [NativeScript/nativescript-cli#5765]
### Issue Description

Starting with NativeScript v8.6.0, `ns prepare ios` may duplicate native code section `TNSNativeSource` in Xcode projects.

In particular, this seems to be related to this PR: https://github.com/NativeScript/nativescript-dev-xcode/pull/15

### Reproduction

1. ns create testApp --js
2. cd testApp
3. ns platform add ios
4. cd App_Resources/iOS
5. mkdir src
6. cd src
7. cp ~/Downloads/extra.zip .
8. unzip extra.zip
9. rm extra.zip
10. cd ../../..
11. ns prepare ios
12. touch App_Resources/iOS/src/extra.m
13. ns prepare ios
14. open platforms/ios/testApp.xcodeproj

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/nativescript-cli/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by ugurkig** [comment]
Having the same issue with NativeScript v8.6.1 and Xcode 14.3

**Comment by rigor789** [comment]
This is known, but should not cause any problems because these are just "links" and not actual copies of the sources. We'd of course like to fix the duplication, but if you are running into build errors it's likely not related to this.

**Comment by ugurkig** [comment]
No, there are no build errors. But it issues warnings against each duplicate native source file and they annoyingly cumulate with every prepare. I have found a workaround to get rid of them like “Delete”ing the duplicate TNSNativeSource entries on Xcode with “Remove References” option, which only deletes the duplicate entry, not the actual files. Thanks for the reply.

**Comment by rigor789** [comment]
This is definitely the right place, it's on NativeScript CLI to fix this. I was just commenting that in general this bug should not cause build issues, but it's definitely a bug.
